pkg/meta/lib/linter.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
```

Comment:
Are users intended to import it directly. If so, it should have a short name, like `meta/lints.dart`.

Why is this file not just in `src/` and exported by `meta.dart`?

Reply:
> Are users intended to import it directly.

Yes.

> Is the `meta` package otherwise separated between lints and "other warnings"?

No, we haven't made such a distinction. As far as I can see there is only one annotation that is related to a lint (`optionalTypeArgs`) and it's of the form that @a14n mentioned elsewhere of an annotation to suppress a lint (a "users can safely violate this lint with this API" semantic).

> Could this file just be in `src/` and be exported by `meta.dart`?

Yes, but there are a couple of reasons why I'm leaning toward making it a separate top-level library:
- This is the first declaration in this package that isn't intended to be used as an annotation, and that feels like a good reason to split it out.
- There are a lot of libraries that import `meta.dart` and I don't want to make breaking changes unnecessarily. Users can import it with a prefix to help prevent collisions when we add a top-level name, but prefixes don't help with extension methods.

Reply:
I'm very unconcerned about extension members colliding. It happens very rarely.
I'd be much more concerned about having to add an extra import, something people will consistently forget to do, then be annoyed and go back and add it (bonus points if we can quick-fix the import, obviously).

I think it would be a better user experience if `import "package:meta/meta.dart";` would import everything you need to interact with the analyzer or linter. Otherwise someone will just introduce `package:metameta/meta.dart` which exports both (I know I would).

Reply:
> I think it would be a better user experience if `import "package:meta/meta.dart";` would import everything you need to interact with the analyzer or linter.

Possibly, but that isn't the way the package is currently organized. Importing `meta.dart` doesn't get you the dart2js-specific annotations, nor the `Target` annotation used only by annotation authors. In those cases we split out the pieces that are not universally useful.

I see this the same way. This extension is only useful to users that have enabled the `unawaited_futures` lint and who don't want to use `// ignore` comments to disable it in specific locations.

Reply:
If we introduce the `unawaited_futures` lint as default in all new Dart projects (and we will), then it does become almost universally useful to have the helper extension.

Otherwise consider having a convenience `package:meta/all.dart` which imports all the (common) annotations and functions/extensions. Maybe everything, or at least everything which isn't platform dependent.
I'd use it (well, if I used `meta` at all).